pkg/i2gw/provider.go

```diff
@@ -105,3 +107,35 @@ type GatewayResources struct {
 // Different FeatureParsers will run in undetermined order. The function must
 // modify / create only the required fields of the gateway resources and nothing else.
 type FeatureParser func([]networkingv1.Ingress, *GatewayResources) field.ErrorList
+
+type ProviderSpecificConf struct {
+	Name         string
+	Description  string
+	DefaultValue string
+}
+
+var (
+	providerSpecificConfs      = map[ProviderName]map[string]ProviderSpecificConf{}
+	providerSpecificConfsMutex = sync.RWMutex{}
+)
+
+// RegisterProviderSpecificConf registers a provider-specific conf.
+// Each provider-specific conf is exposed to the user as a command-line flag, defined as optional.
+// If the flag is not provided, it is up for the provider to decide to use the default value or raise an error.
+// The provider can read the values of provider-specific confs input by the user from the ProviderConf,
+// prefixed by the provider name.
```

Comment:
Thanks, can you add a line, explaining why you use the lock here. I want to save other folks time and searching the codebase for concurrency.